src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -906,6 +897,113 @@ static bool usesReceiver(BoundExpression argument)
             }
         }
 
+        private RefKind GetRefKindForCapturedReceiverTemp(BoundExpression rewrittenReceiver)
+        {
+            Debug.Assert(rewrittenReceiver.Type is not null);
+
+            RefKind refKind = rewrittenReceiver.GetRefKind();
+            if (refKind == RefKind.RefReadOnlyParameter)
+            {
+                refKind = RefKind.Ref;
+            }
+            else if (refKind == RefKind.None &&
+                !rewrittenReceiver.Type.IsReferenceType &&
+                Binder.HasHome(rewrittenReceiver,
+                               Binder.AddressKind.Constrained,
+                               _factory.CurrentFunction,
+                               peVerifyCompatEnabled: false,
+                               stackLocalsOpt: null))
+            {
+                refKind = RefKind.Ref;
+            }
+
+            return refKind;
+        }
+
+        // This may introduce some temps into an outer scope. The caller has to deal with that.
+        [return: NotNullIfNotNull(nameof(receiver))]
+        private BoundExpression? AdjustReceiverForExtensionsIfNeeded(BoundExpression? receiver, Symbol member,
+            ArrayBuilder<BoundExpression>? storesOpt, [NotNullIfNotNull(nameof(tempsOpt))] ref ArrayBuilder<LocalSymbol>? tempsOpt)
+        {
+            if (receiver is null)
+            {
+                return receiver;
+            }
+
+            Debug.Assert(member is not null);
+            if (receiver is BoundTypeExpression)
+            {
+                return receiver;
+            }
+
+            if (receiver is BoundThisReference)
+            {
+                // PROTOTYPE this was only minimally tested to observe side-effects but needs more work (in CheckValue for instance, interpolation handlers)
```

Comment:
>interpolation handlers [](http://example.com/codeflow?start=133&length=22)

Has this part been handled?

Reply:
The whole section of handling `this.` adjustment is under PROTOTYPE comment above. It is not meant to be complete, just minimally allow observing side-effects for this PR. In other words, this is not in scope for this PR